react/features/base/tracks/actionTypes.js

```diff
@@ -10,15 +10,45 @@
 export const TRACK_ADDED = Symbol('TRACK_ADDED');
 
 /**
- * Action for when a track cannot be created because permissions have not been
- * granted.
+ * Action triggered when a local track starts being created through the WebRTC
+ * getUserMedia call. It will include extra 'gumProcess' field which is
+ * a Promise with extra 'cancel' method which can be used to cancel the process.
+ * Canceling will result in disposing any JitsiLocalTrack returned by the GUM
+ * callback. There will be TRACK_CREATE_CANCELED event instead of track
+ * added/gum failed events.
  *
- * {
- *     type: TRACK_PERMISSION_ERROR,
- *     trackType: string
+ * @type {
+ *      type: TRACK_BEING_CREATED
+ *      track: {
+ *          local: true,
+ *          mediaType: MEDIA_TYPE,
+ *          gumProcess: Promise with cancel() method to abort
+ *      }
+ * }
+ */
+export const TRACK_BEING_CREATED = Symbol('TRACK_BEING_CREATED');
+
+/**
+ * Action sent when canceled GUM process completes either successfully or with
+ * an error (error is ignored and track is immediately disposed if created).
+ *
+ * @type {
+ *      type: TRACK_CREATE_CANCELED,
```

Comment:
I see this one often so I guess it's some kind of editor/IDE-induced indentation. The story goes like this:

1. The line begins with ` * ` because that's how we format the multiline jsdoc comment. Afterwards, we put the content of the comment. Everything's as it should be here.

2. Then we've got 5 more spaces which indent the properties of the class/object. But that's not consistent with our indentation which requires 4 spaces. Could you please use 4 spaces per indent?

Reply:
yes, it is an IDE thing - I'll try to pay bigger attention to that

Reply:
also in some places I was putting 'type:' first before any other properties even if it's not in alphabetical order, because it feels like a label and makes it easier to read an action type. Is that ok ?

Reply:
Yes, @paweldomas, that's exactly what we do for action payloads everywhere.